### PR TITLE
Added lua

### PR DESCRIPTION
**Pull request by robindiddams**
I added lua comment parsing and a popular unit testing framework "busted", along with tests for both

### CODE REVIEW

File: pkg/lua/lua.go [hunk 1]
```diff
@@ -2,13 +2,18 @@ package lua
 
 import (
 	"github.com/pinpt/dialect"
+	"github.com/pinpt/dialect/pkg/lua/busted"
```
Comment (jhaynie) [inline pkg/lua/lua.go:5]: busted? is this old?

File: pkg/lua/busted/busted.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+)
+
+var (
+	containsDescribe = regexp.MustCompile("describe\\s*\\(")
+	containsIt       = regexp.MustCompile("it\\s?(\\.only|skip)?\\s?\\(")
+)
+
+func IsTest(line *dialect.DialectLine) bool {
+	return containsDescribe.MatchString(line.Buffer) && containsIt.MatchString(line.Buffer)
+}
```
Comment (jhaynie) [inline pkg/lua/busted/busted.go:15]: i think these was left over from javascript

**Review by jhaynie — CHANGES_REQUESTED** [review]
i think you have some issues with `busted` that need to be resolved.

File: pkg/lua/lua.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	"strings"
+)
+
+type LuaExaminer struct {
+	inDoubleComment bool
+}
+
+func isTest(line *dialect.DialectLine) bool {
+	return busted.IsTest(line)
+}
```
Comment (jhaynie) [inline pkg/lua/lua.go:15]: is this old?

File: pkg/lua/busted/busted.go (hunk 2 above)
Comment (robindiddams) [reply pkg/lua/busted/busted.go:15]: busted uses the same syntax as mocha, so I used the same code from mocha. or am I missing your question?

Comment (jhaynie) [reply pkg/lua/busted/busted.go:15]: oh so busted is the name of a lua testing package? oops. i thought it was some weird name of your pkg. 👍

**Pull request merged**